Updated menu system, updated sound effects, tweaks and updated STL's.

gpstar81 opened the pull request:
-New Afterlife sound effects and various sound effect tweaks.
-Updated STL's.
-New menu options added to the menu system.

Yanks1023 commented:
Hi Michael,I haven’t updated my diy software since before you put the 3 led option. I’m interested in getting updated sounds but are there huge changes to the menu system? I would need to update both my sd cards and reflash both the nano and mega, correct? I do not have the bar graph so hopefully that’s not an issue with updating code. I have it at a spot I’m happy with and don’t want to change much. Updated sound fx would be something I’d want though if they are significantly improved. Thanks again for the support! -Justin On Jul 21, 2023, at 9:58 AM, Michael Rajotte ***@***.***> wrote:﻿-New Afterlife sound effects and various sound effect tweaks.
-Updated STL's.
-New menu options added to the menu system.



You can view, comment on, or merge this pull request online at:
  https://github.com/gpstar81/haslab-proton-pack/pull/106

Commit Summary

  c8c9fbc  Wand menu adjustments, brightness control of the pack lights, updated sound effects, stls, etc
  4099239  Wand menu adjustments, brightness control of the pack lights, updated sound effects, stls, etc
  20e587f  various fixes, updated stls


File Changes  (51 files)

  
    M
    ADVCONFIG.md
    (11)
  
  
    M
    OPERATION.md
    (8)
  
  
    M
    sounds/001_bootup.wav
    (0)
  
  
    M
    sounds/003_idle_loop.wav
    (0)
  
  
    M
    sounds/015_afterlife_pack_startup.wav
    (0)
  
  
    M
    sounds/016_afterlife_pack_idle_loop.wav
    (0)
  
  
    M
    sounds/036_pack_shutdown_afterlife.wav
    (0)
  
  
    M
    sounds/038_gb2_pack_loop.wav
    (0)
  
  
    M
    sounds/102_proton_pack_vibration_enabled.wav
    (0)
  
  
    M
    sounds/103_proton_pack_vibration_disabled.wav
    (0)
  
  
    M
    sounds/104_proton_pack_vibration_firing_enabled.wav
    (0)
  
  
    M
    sounds/105_proton_pack_vibration_firing_disabled.wav
    (0)
  
  
    M
    sounds/106_neutrona_wand_vibration_enabled.wav
    (0)
  
  
    M
    sounds/107_neutrona_wand_vibration_disabled.wav
    (0)
  
  
    M
    sounds/108_neutrona_wand_vibration_firing_enabled.wav
    (0)
  
  
    M
    sounds/109_neutrona_wand_vibration_firing_disabled.wav
    (0)
  
  
    M
    sounds/112_video_game_colours_disabled.wav
    (0)
  
  
    M
    sounds/113_video_game_colours_enabled.wav
    (0)
  
  
    M
    sounds/114_video_game_colours_powercell_enabled.wav
    (0)
  
  
    M
    sounds/115_video_game_colours_cyclotron_enabled.wav
    (0)
  
  
    A
    sounds/116_fire_sparks.wav
    (0)
  
  
    A
    sounds/117_fire_sparks_2.wav
    (0)
  
  
    A
    sounds/118_fire_sparks_3.wav
    (0)
  
  
    A
    sounds/119_fire_sparks_4.wav
    (0)
  
  
    A
    sounds/120_fire_sparks_5.wav
    (0)
  
  
    A
    sounds/121_voice_powercell_brightness.wav
    (0)
  
  
    A
    sounds/122_voice_cyclotron_brightness.wav
    (0)
  
  
    A
    sounds/123_voice_cyclotron_inner_brightness.wav
    (0)
  
  
    A
    sounds/124_voice_proton_mix_effects_enabled.wav
    (0)
  
  
    A
    sounds/125_voice_proton_mix_effects_disabled.wav
    (0)
  
  
    A
    sounds/126_alarm_loop.wav
    (0)
  
  
    A
    sounds/127_ribbon_cable_start.wav
    (0)
  
  
    M
    source/NeutronaWand/Communication.h
    (14)
  
  
    M
    source/NeutronaWand/Configuration.h
    (10)
  
  
    M
    source/NeutronaWand/Header.h
    (15)
  
  
    M
    source/NeutronaWand/MusicSounds.h
    (16)
  
  
    M
    source/NeutronaWand/NeutronaWand.ino
    (279)
  
  
    M
    source/ProtonPack/Communication.h
    (12)
  
  
    M
    source/ProtonPack/Configuration.h
    (30)
  
  
    M
    source/ProtonPack/Header.h
    (18)
  
  
    M
    source/ProtonPack/MusicSounds.h
    (16)
  
  
    M
    source/ProtonPack/ProtonPack.ino
    (764)
  
  
    M
    stl/pack/Speaker Mount V2/Bottom_Speaker_Bracket.stl
    (0)
  
  
    M
    stl/pack/Speaker Mount V2/Top_Speaker_Mount_Bracket.stl
    (0)
  
  
    R
    stl/wand/Wand Connector/V1_Wand_Connector_Blank.stl
    (0)
  
  
    R
    stl/wand/Wand Connector/V1_Wand_Connector_Hole_Bottom.stl
    (0)
  
  
    R
    stl/wand/Wand Connector/V1_Wand_Connector_Hole_Left.stl
    (0)
  
  
    A
    stl/wand/Wand Connector/V2_Wand_Connector_Blank.stl
    (0)
  
  
    A
    stl/wand/Wand Connector/V2_Wand_Connector_Hole_Bottom.stl
    (0)
  
  
    A
    stl/wand/Wand Connector/V2_Wand_Connector_Hole_Left.stl
    (0)
  
  
    M
    stl/wand/Wand Wire Box/v2_wire_box_risers.stl
    (0)
  


Patch Links:

  https://github.com/gpstar81/haslab-proton-pack/pull/106.patch
  https://github.com/gpstar81/haslab-proton-pack/pull/106.diff


—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

gpstar81 commented:
> Hi Michael,I haven’t updated my diy software since before you put the 3 led option. I’m interested in getting updated sounds but are there huge changes to the menu system? I would need to update both my sd cards and reflash both the nano and mega, correct? I do not have the bar graph so hopefully that’s not an issue with updating code. I have it at a spot I’m happy with and don’t want to change much. Updated sound fx would be something I’d want though if they are significantly improved. Thanks again for the support! -Justin >

Hi Justin,

Yes you can update. You will need to reflash both your Mega and Nano and replace all the files on the SD cards. Also take note, music tracks are now starting on track 500, so old music tracks will need to be renamed and adjusted.

Also for the Nano, there is one more extra step required to flash it, refer to flashing guide for the DIY builders here:
https://github.com/gpstar81/haslab-proton-pack/blob/main/ARDUINO_FLASHING.md

Note the new library added: SerialTransfer that you need to install. 
I believe you have a original wiring layout on your Mega, so you need to refer to the Configuration.h in the Pack code at the very bottom and comment out the last line (There is a explaination at the bottom of that file explaining why and if required or not).

Also remember, the old version of the code has been archived in the legacy-diy branch and you can always revert back.

DustinGrau commented on source/ProtonPack/ProtonPack.ino:
```diff
@@ -1121,6 +1231,114 @@ void powercellOff() {
+  for(int i = 0; i <= i_powercell_led; i++) {
+    uint8_t r = 0;
+    uint8_t g = 0;
+    uint8_t b = 0 + (255 * i_powercell_brightness / 100);
```
This calculation appears quite a bit now. Recommend making a helper function to do this adjustment. Conversely, if the value might be calculated when the dimming adjustment is made, that value could be a global and simply used in each location where this calculation is performed (it's minor, but every math operation adds up given the number of times the LED's transition).

gpstar81 replied:
Cleaned it up a bit now. Only appearing once now.

DustinGrau commented on source/ProtonPack/ProtonPack.ino:
```diff
@@ -283,8 +296,76 @@ void loop() {
+            break;
+          }
+
+          unsigned int i_s_random = random(2,4) * 1000;
```
Is there benefit of making this range a Configuration.h option to affect how wide the range of randomness might be? Also, is this essentially setting a time between 2-4 seconds?

gpstar81 replied:
Yes between 2-4 seconds, but sometimes 12-14 at times. If set in the configuration then it wont be random on each cycle.

DustinGrau reviewed (changes requested):
Overall I focused on the code changes which look good. Only one true concern regarding some repeated calculations for the pack and dimming.

Yanks1023 commented:
Okay, I’m looking in the most recent code (ProtonPack.ino) for the Config.h line and the explanation. Im sorry, I don’t see it. I see #include “Configuration.h” at the beginning tho. Which line is that config.h that needs to be commented out? On Jul 21, 2023, at 11:07 AM, Michael Rajotte ***@***.***> wrote:﻿

Hi Michael,I haven’t updated my diy software since before you put the 3 led option. I’m interested in getting updated sounds but are there huge changes to the menu system? I would need to update both my sd cards and reflash both the nano and mega, correct? I do not have the bar graph so hopefully that’s not an issue with updating code. I have it at a spot I’m happy with and don’t want to change much. Updated sound fx would be something I’d want though if they are significantly improved. Thanks again for the support! -Justin >

Hi Justin,
Yes you can update. You will need to reflash both your Mega and Nano and replace all the files on the SD cards. Also take note, music tracks are now starting on track 500, so old music tracks will need to be renamed and adjusted.
Also for the Nano, there is one more extra step required to flash it, refer to flashing guide for the DIY builders here:
https://github.com/gpstar81/haslab-proton-pack/blob/main/ARDUINO_FLASHING.md
Note the new library added: SerialTransfer that you need to install.
I believe you have a original wiring layout on your Mega, so you need to refer to the Configuration.h in the Pack code at the very bottom and comment out the last line (There is a explaination at the bottom of that file explaining why and if required or not).
Also remember, the old version of the code has been archived in the legacy-diy branch and you can always revert back.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>

gpstar81 commented:
> 

Oh, open Configuration.h from the Proton Pack project. The very bottom of the file. You will find:

#define GPSTAR_PROTON_PACK_PCB

Yanks1023 commented:
Found it. Thank you! -JOn Jul 21, 2023, at 11:40 AM, Michael Rajotte ***@***.***> wrote:﻿


Oh, open Configuration.h from the Proton Pack project. The very bottom of the file. You will find:
#define GPSTAR_PROTON_PACK_PCB

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>

gpstar81 commented:
> The pack setup (diy) shows the lid detection as pin 43. I can’t recall but I think that’s how I have mine wired, so according to your notes, I can leave that alone. -J On Jul 21, 2023, at 11:40 AM, Michael Rajotte ***@***.***> wrote:﻿ Oh, open Configuration.h from the Proton Pack project. The very bottom of the file. You will find: #define GPSTAR_PROTON_PACK_PCB —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>

It is 43 now, but the older plans were at 51, you may be on 51. Either way, if your lid detection does not work, then you are on 51 as per the older seutp.

DustinGrau reviewed (approved):
Calling this one good, and will do testing in the near future
